data/trie/branchNode.go

```diff
@@ -779,6 +779,30 @@ func (bn *branchNode) getAllHashes(db data.DBWriteCacher) ([][]byte, error) {
 	return hashes, nil
 }
 
+func (bn *branchNode) getNumNodes() data.NumNodesDTO {
+	arg := data.NumNodesDTO{
+		Total:    1,
+		Branches: 1,
+	}
+
+	for _, n := range bn.children {
+		if n != nil {
+			arg2 := n.getNumNodes()
+			arg.Total += arg2.Total
```

Comment:
I know but then this will become a DTO with logic which I'm not really happy to do it. Since that DTO is used in an interface definition I would keep the structure as lightweight as possible